lib/src/model.ts

```diff
@@ -63,7 +64,16 @@ export class Model extends EventTarget {
         this._ready = message.result;
       }
 
-      this.dispatchEvent(new CustomEvent(message.event, { detail: message }));
+      const event = new CustomEvent(message.event, { detail: message, ...message });
+      if (ServerMessage.isRecognizerMessage(message) && message.recognizerId) {
+        const recognizer = this.recognizers.get(message.recognizerId);
+        if (recognizer) {
+          recognizer.dispatchEvent(event);
```

Comment:
I think if you dispatch specific recognizer events via the recognizer here you don´t need the "on" method in the KaldiRecognizer class at all, don´t you?

3rd party code can add an event listener to the recognizer and will receive the event naturally.